iree/compiler/Dialect/Flow/IR/FlowOps.td

```diff
@@ -899,25 +972,33 @@ def FLOW_TensorUpdateOp : FLOW_PureOp<"tensor.update", [
   }];
 
   let arguments = (ins
-    FLOW_Tensor:$update,
     FLOW_Tensor:$target,
-    Variadic<FLOW_Dim>:$start_indices
+    FLOW_DynamicDims:$target_dims,
+    Variadic<FLOW_Dim>:$start_indices,
```

Comment:
Should this use `FLOW_DynamicDims`? If not, then adding a comment on FLOW_DynamicDims explaining the intended usage (rather than just as a macro for Variadict<FLOW_Dim>) would be useful

Reply:
May be a call for a better name; the DynamicDims are meant to represent only dynamic dimensions - whereas start_indices here are all of the start indices. Maybe `FLOW_ShapeDynamicDims` - I'll go with that for now but happy to hear suggestions!